pkg/flowaggregator/s3uploader/s3uploader.go

```diff
@@ -0,0 +1,638 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3uploader
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	s3service "github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/gammazero/deque"
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/flowaggregator/flowrecord"
+)
+
+var (
+	buffer        bytes.Buffer
+	awsS3Client   *s3service.Client
+	awsS3Uploader *s3manager.Uploader
+	s3Uploader    S3Uploader
+)
+
+const (
+	queueFlushTimeout = 1 * time.Minute
+)
+
+type stopPayload struct {
+	flushQueue bool
+}
+
+type S3ExportProcess struct {
```

Comment:
not sure why this is called `S3ExportProcess` when the file is called `s3uploader.go`